### DISCUSSION TITLE
REM and MQTT

### DISCUSSION THREAD

**Discussion by orrious** [tagyoureit/nodejs-poolController#564] [REM]
I've tried unsuccessfully to setup REM with MQTT directly and am not able to find a guide where someone else has done it.  Can you share how you configure a REM relay to be controlled via MQTT?

I've created a MQTT connection in General.
I went into the I2C Multi-Relay Board: 
    Triggers:
        Connection: mqtt
        Set Value: relay state
        Relay: 1
        Topic: rem/relay1/set
    Feeds:
        Connection: mqtt
        Send Value: Value for relay #1
        Topic: rem/relay1

I'm assuming I need something in the javascript fields

**Comment by rstrouse** [comment]
Please pull REM.  The mqtt client was not re-subscribing to the topic after the trigger was saved.  The mqtt client changed the returns so that the promise was not being resolved.  This meant that once you saved the trigger it would unsubscribe but never re-subscribe.

You do not need to do anything with the filters or binding if the contents of your set looks like the following.

After you set this up you can set any one of the following as a value.
`0 or 1`
`{"state": true | false}`
`{"state": true, latch: number}`  where the latch is the number of ms to hold onto the latch.  If another set is not sent in that time it will release.
`[{"state":true, "timeout": 300}, {"state":false, "timeout":300}, {"state":true, "timeout":300}]` will turn the relay on-off-on with 300ms in between.
